### PR TITLE
Feature/KAH-6 ( Settings endpoints tests )

### PR DESCRIPTION
**Pull request by arispen**
Test Suites: **14 passed**, 14 total
Tests:       **144 passed**, 144 total

### CODE REVIEW

File: server/api/tests/settings.test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        });
+
+        it(`should respond with 500 for error connecting to db`, () => {
+            return request(app)
+                .post('/api/settings/db')
+                .send({ uri: 'mongodb+srv://admin:wrong-password@cluster0-qnjcz.mongodb.net/jest-settings-test' })
+                .expect(500);
+        });
+    });
+});
```
Comment (ChBernat) [inline server/api/tests/settings.test.js]: Add a new line at the end of the file, please.

File: server/api/tests/settings.test.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                    });
+            });
+        });
+    });
+
+    describe('POST api/settings/db', () => {
+        it(`should respond with 204 after successful connection to given db uri`, () => {
+            return request(app)
+                .post('/api/settings/db')
+                .send({ uri: 'mongodb+srv://admin:7vXYF4nNu8B9SnH5@cluster0-qnjcz.mongodb.net/jest-settings-test' })
```
Comment (theflexbird) [inline server/api/tests/settings.test.js]: Use only local db.

File: server/api/tests/settings.test.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                .expect(500)
+                .then(({ text }) => {
+                    expect(text).toEqual('Missing parameters');
+                });
+        });
+
+        it(`should respond with 500 for error connecting to db`, () => {
+            return request(app)
+                .post('/api/settings/db')
+                .send({ uri: 'mongodb+srv://admin:wrong-password@cluster0-qnjcz.mongodb.net/jest-settings-test' })
```
Comment (theflexbird) [inline server/api/tests/settings.test.js]: Use only local db.

File: server/api/tests/settings.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
+const request = require('supertest');
+const fs = require('fs');
+const path = require('path');
+
+const app = 'localhost:3000';
+let originalConfig = {};
+const testConfig = { "upload_path": "uploads", "interval_time": 5000, "retries": 3, "page_size": 15, "dbURI": "mongodb+srv://admin:7vXYF4nNu8B9SnH5@cluster0-qnjcz.mongodb.net/jest-settings-test", "serverKey": "b92a5c991d68e544a3fa21da9b083903" }
```
Comment (theflexbird) [inline server/api/tests/settings.test.js]: Use only local db. Replace `mongodb+srv://admin:7vXYF4nNu8B9SnH5@cluster0-qnjcz.mongodb.net/jest-settings-test`

**Pull request merged**